cmake/OpenEXRSetup.cmake

```diff
@@ -160,15 +160,47 @@ set(OPENEXR_DEFLATE_TAG "v1.18" CACHE STRING "Tag to use for libdeflate source r
 if(NOT OPENEXR_FORCE_INTERNAL_DEFLATE)
   #TODO: ^^ Release should not clone from main, this is a place holder
   set(CMAKE_IGNORE_PATH "${CMAKE_CURRENT_BINARY_DIR}/_deps/deflate-src/config;${CMAKE_CURRENT_BINARY_DIR}/_deps/deflate-build/config")
-  include(FindPkgConfig)
-  pkg_check_modules(deflate IMPORTED_TARGET GLOBAL libdeflate)
-  set(CMAKE_IGNORE_PATH)
-  if (deflate_FOUND)
-    message(STATUS "Using libdeflate from ${deflate_LINK_LIBRARIES}")
+  # First try cmake config
+  find_package(libdeflate CONFIG)
+  if(libdeflate_FOUND)
+    if(TARGET libdeflate::libdeflate_shared)
+      set(EXR_DEFLATE_LIB libdeflate::libdeflate_shared)
+    else()
+      set(EXR_DEFLATE_LIB libdeflate::libdeflate_static)
+    endif()
+    set(EXR_DEFLATE_SOURCES)
+    set(EXR_DEFLATE_INCLUDE_DIR)
+    set(EXR_DEFLATE_LDFLAGS)
+    # Set EXR_DEFATE_LDFLAGS for OpenEXR.pc.in for static build
+    if(NOT BUILD_SHARED_LIBS)
+      get_target_property(deflate_LIBRARIES ${EXR_DEFLATE_LIB} LOCATION)
+      set(EXR_DEFLATE_LDFLAGS "-l${deflate_LIBRARIES}")
```

Comment:
libdeflate ships a .pc file, so this should actually go in `Requires.private` as `libdeflate`; then it doesn't even have to be gated by `if(NOT BUILD_SHARED_LIBS)`